### PR TITLE
feat: add multiple file support

### LINKED ISSUE #2: multifile and crossfile references
there are cases when you have several spec files. it causes need to proceed several files in one go. and there might be references to other files. we need to support such cases

### PR DESCRIPTION
**Pull request by scink**
Closes #2

BREAKING CHANGE: `pathToSpec` renamed to `pathsToSpec` and it is `string[]` now instead of `string`

### CODE REVIEW

**Comment by scink** [comment]
@raveclassic, please review this PR

File: src/language/typescript.ts [hunk 1]
```diff
@@ -218,10 +220,29 @@ const serializePath = (url: string, item: TPathItemObject, rootName: string, cwd
 const serializeSchemaObject = (schema: TSchemaObject, rootName: string, cwd: string): TSerializedType => {
 	switch (schema.type) {
 		case undefined: {
-			const type = `${schema.$ref.replace(/^#\/definitions\//g, '')}`;
+			const $ref = schema.$ref;
+			const defBlock = fromNullable($ref.match(/#\/[^\/]+\//))
```
Comment (raveclassic) [inline src/language/typescript.ts]: What are you trying to achieve here? I believe everything can be rewritten with a single match

File: test/specs/common.json [hunk 2]
```diff
@@ -0,0 +1,182 @@
+{
```
Comment (raveclassic) [inline test/specs/common.json]: Default identation for json files is 2 spaces. See .editorconfig

File: package.json [hunk 3]
```diff
@@ -6,7 +6,7 @@
   "scripts": {
     "ts-node": "ts-node",
     "tslint": "tslint -c tslint.json --project tsconfig.json './src/**/*.ts' './test/**/*.ts'",
-    "test": "npm run tslint && npm run prettier && ts-node ./test/index.ts",
+    "test": "rm -rf ./test/out && npm run tslint && npm run prettier && ts-node ./test/index.ts",
```
[the remark this answers is not in the document]
Comment (scink) [reply package.json]: i used `del`

File: src/index.ts [hunk 4]
```diff
@@ -28,42 +37,82 @@ export type TGenerateOptions = {
...
+
+const read = async (_pathToFile: string): Promise<TBuffer> => {
+	const pathToFile = resolvePath(_pathToFile);
+	return {
+		buffer: await fs.readFile(pathToFile),
+		fileName: path.basename(pathToFile),
+	};
+};
+
+const decode = (json: TJSON): Either<ValidationError[], TSwaggerObject> => {
```
[the remark this answers is not in the document]
Comment (scink) [reply src/index.ts]: done

File: src/index.ts [hunk 5]
```diff
@@ -28,42 +37,82 @@ export type TGenerateOptions = {
...
-	log('Serializing spec');
-	const serialized = serialize(path.basename(out), decoded.value);
-	log('Running prettier');
-	const prettierConfig = fromNullable(
+type TBuffer = {
+	buffer: Buffer;
+	fileName: string;
+};
+
+const read = async (_pathToFile: string): Promise<TBuffer> => {
```
[the remark this answers is not in the document]
Comment (scink) [reply src/index.ts]: done

File: test/specs/common.json (hunk 2 above)
Comment (scink) [reply test/specs/common.json]: there is nothing about `json` in the `.editorconfig`

File: src/index.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (scink) [reply src/index.ts]: deleted

File: src/index.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (scink) [reply src/index.ts]: done

File: src/language/typescript.ts (hunk 1 above)
Comment (scink) [reply src/language/typescript.ts]: replaced with a single match

**Pull request merged**